### PR TITLE
stop returning NS records on authoritative response

### PR DESCRIPTION
**Pull request by bluejekyll**
This disables NS records from being included in the response on Authoritative responses.

Fixes: #1531

### CODE REVIEW

**Comment by djc** [comment]
Presumably you actually want to remove that code rather than commenting it out?

It would be good to document why we're not returning NS records, if that's in an RFC somewhere or if some authoritative reference says we shouldn't do that... Or, failing that, just some comment explaining the issue here.

**Comment by bluejekyll** [comment]
See the related issue. This is an investigation to see if it will resolve an incompatibility with the way authoritative records are being returned. If it's successful, then I will absolutely update the PR to be correct. It's partly why it's in a draft state.

**Comment by bluejekyll** [comment]
@williamdes ok, I just pushed some changes (that will need to be cleaned up before merge) that should enforce better the `rd` bits and make sure the `rd` is set on the response based on the request.

**Review by williamdes — APPROVED** [review]
🎉 🚀 
You rock !
https://github.com/bluejekyll/trust-dns/issues/1531#issuecomment-900214795

**Comment by bluejekyll** [comment]
Ok, I think this is ready for merging. @williamdes, it would be great if you have some time to very the new changes, just to make sure I didn't break anything while cleaning this up.

**Comment by williamdes** [comment]
> Ok, I think this is ready for merging. @williamdes, it would be great if you have some time to very the new changes, just to make sure I didn't break anything while cleaning this up.

The code looks good, I did see two things commented to be done but nothing more
When everything is okay I can probably fire this into production and run their test tool just in case

**Comment by bluejekyll** [comment]
oh, I missed that test, thanks for catching that.

**Comment by bluejekyll** [comment]
ok, I got rid of that last bit of commented out code.

File: bin/tests/named_tests.rs [hunk 1]
```diff
@@ -282,6 +282,7 @@ fn test_server_continues_on_bad_data_tcp() {
 
 #[test]
 #[cfg(feature = "resolver")]
+#[ignore] // FIXME: the client needs to be updated to specify that recursion is desired on the request... before commit
```
Comment (williamdes) [inline bin/tests/named_tests.rs:285]: what about this one ?

Comment (bluejekyll) [reply bin/tests/named_tests.rs:285]: Good catch. Not sure how I missed that.

**Comment by bluejekyll** [comment]
I'm happy you caught that, because that test detected a bug in the changes... fixed now.

**Review by williamdes — APPROVED** [review]
Looks good, will let you know when I deploy it

**Comment by bluejekyll** [comment]
@williamdes, any update on being able to confirm this?

**Comment by williamdes** [comment]
> @williamdes, any update on being able to confirm this?

Hi,

Sorry for the delay, re-built, published, and deployed the image into production 

And it works, I removed and added back one name server and could see that the requests where handled correctly.
https://apps.db.ripe.net/db-web-ui/lookup?source=RIPE&key=4.1.0.0.6.4.6.4.0.1.a.2.ip6.arpa&type=domain

🚢 🚀

**Comment by bluejekyll** [comment]
Excellent. Thank you for the review and helping track down the issues to get this patch in!

Merging.

**Pull request merged**